monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/RecursiveConcatSuite.scala

```diff
@@ -48,6 +48,17 @@ object RecursiveConcatSuite extends BaseOperatorSuite {
   test("stack safety") { implicit s =>
     val count = 10000
     val f = range(0, count).sumL.runAsync; s.tick()
-    assertEquals(f.value, Some(Success(count.toLong * (count - 1) / 2)))
+    assertEquals(f.value, Some(Success(sum(count))))
+  }
+
+  val nats: Observable[Long] = {
+    Observable.now(1L) ++ nats.map(_ + 1)
```

Comment:
This PR is really old, sorry for coming back so late. I'm not a good PR manager 😐

Anyway, so one problem is that this `nats` definition is not _tail recursive_, therefore the application of `++` cannot be. Actually in this case that `Observable` throws `StackOverflowException` is a feature, because otherwise you have a memory leak on your hands.

A tail-recursive `nats` definition would go like this:

```scala
val nats: Observable[Long] = {
  def loop(acc: Long): Observable[Long] =
    Observable.now(acc) ++ loop(acc + 1)
  loop(1)
}
```